src/jshint.js

```diff
@@ -1340,8 +1340,10 @@ var JSHINT = (function() {
       while (!obj.identifier && typeof obj.left === "object")
         obj = obj.left;
 
-      if (obj.identifier && natives.indexOf(obj.value) >= 0)
+      if (obj.identifier && natives.indexOf(obj.value) >= 0 &&
+          state.funct["(scope)"].isPredefined(obj.value)) {
```

Comment:
I think it is better to check if the variable is predefined in the `findNativePrototype` function so that it returns something only if you are actually modifying a native prototype. (which is the behaviour suggested by its name)

Anyway, this is a very little difference: feel free to go ahead and move it to `checkLeftAssign`